x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/history/ILMHistoryStore.java

```diff
@@ -120,10 +113,13 @@ public void afterBulk(long executionId, BulkRequest request, BulkResponse respon
                                 .collect(Collectors.joining(",")));
                     }
                     if (response.hasFailures()) {
-                        Map<String, String> failures = Arrays.stream(response.getItems())
+                        IllegalStateException e = new IllegalStateException("failures during bulk request");
+                        Arrays.stream(response.getItems())
                             .filter(BulkItemResponse::isFailed)
-                            .collect(Collectors.toMap(BulkItemResponse::getId, BulkItemResponse::getFailureMessage));
-                        logger.error("failures: [{}]", failures);
+                            .map(r->r.getFailure().getCause())
```

Comment:
I'm not sure we should change the way we report errors. eg. could the failure cause be null ever? if so `addSuppressed` would fail. I think we should keep the `BulkItemResponse#getFailureMessage` as the reported error. 
What do you think?